File: src/main/java/destiny/astrology/SolarEclipseTotal.java
/**
 * Created by smallufo on 2017-11-05.
 */
package destiny.astrology;

public class SolarEclipseTotal extends AbstractSolarEclipse {

  protected final double totalBegin;

  protected final double totalEnd;

  public SolarEclipseTotal(double max, double partialBegin, double partialEnd, double totalBegin, double totalEnd) {
    super(max, partialBegin , partialEnd);
    this.totalBegin = totalBegin;
    this.totalEnd = totalEnd;

  }

  public double getTotalBegin() {
    return totalBegin;
  }

  public double getTotalEnd() {
    return totalEnd;
  }

  @Override
  public SolarType getSolarType() {
    return SolarType.TOTAL;
  }

  @Override
  public String toString() {
    return "[全食 " +
      "begin=" + begin +
      ", totalBegin=" + totalBegin +
      ", max=" + max +
      ", totalEnd=" + totalEnd +
      ", end=" + end + ']';
  }

}

File: src/main/java/destiny/astrology/eclipse/SolarEclipseTotal.kt
/**
 * Created by smallufo on 2017-11-05.
 */
package destiny.astrology.eclipse

open class SolarEclipseTotal(max: Double,
                             partialBegin: Double,
                             partialEnd: Double,
                             val totalBegin: Double,
                             val totalEnd: Double) : AbstractSolarEclipse(max, partialBegin, partialEnd) {

  override val solarType: ISolarEclipse.SolarType
    get() = ISolarEclipse.SolarType.TOTAL

  override fun toString(): String {
    return "[全食 " +
      "begin=" + begin +
      ", totalBegin=" + totalBegin +
      ", max=" + max +
      ", totalEnd=" + totalEnd +
      ", end=" + end + ']'
  }

}
